### PR TITLE
chore: Add missing dependencies

### PR DESCRIPTION
**Pull request by scinos**
#### Changes proposed in this Pull Request

Adds dependencies to package that were used but not declared.

### CODE REVIEW

**Comment by noahtallen** [comment]
Will we be able to enable a rule in the future which prevents us from merging undeclared dependencies?

**Comment by scinos** [comment]
> Will we be able to enable a rule in the future which prevents us from merging undeclared dependencies?

I haven't found any reliable tool or strategy :( Even when the script I wrote (which detects more usages than `depcheck`, I still got lots of false positives, for example when a package name appears in a comment.

**Comment by noahtallen** [comment]
Yeah... I wonder if we could start fairly narrow (like, a line which matches `^import anything from 'anything';`)

**Review by noahtallen — APPROVED** [review]
LGTM, as long as the e2e tests are unrelated

**Comment by scinos** [comment]
> Yeah... I wonder if we could start fairly narrow (like, a line which matches ^import anything from 'anything';)

That's a good point, we can start there. I think `depcheck` already covers that, so we don't have to write anything.

Another complementary idea is to use focused workspaces. In theory, you should be able to build any package in our repo with `yarn workspaces focus <package> && yarn workspace <package> build`. If not, that's a strong signal that the package is using undeclared dependencies.

File: packages/design-picker/package.json [hunk 1]
```diff
@@ -42,13 +43,21 @@
...
 		"@testing-library/jest-dom": "^5.15.0",
 		"@testing-library/react": "^12.1.2",
+		"enzyme": "^3.11.0",
+		"jest": "^27.2.4",
 		"jest-canvas-mock": "^2.3.0",
+		"postcss": "^8.3.11",
 		"react": "^17.0.2",
 		"react-dom": "^17.0.2",
-		"typescript": "^4.4.4"
+		"react-native": "^0.66.2",
```
Comment (noahtallen) [inline packages/design-picker/package.json:56]: Do you know why this was needed? I plan to remove it in https://github.com/Automattic/wp-calypso/pull/58228. I couldn't find any other use of `react-native` in the codebase -- maybe stale from when we still had it around?

**Pull request merged**